In [35]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

#Reading the data set

In [38]:
df=pd.read_csv('StudentsPerformance.csv')

In [15]:
df.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

Checking for null values

In [23]:
df.isnull()

gender  race/ethnicity  parental level of education  lunch  \
0     False           False                        False  False   
1     False           False                        False  False   
2     False           False                        False  False   
3     False           False                        False  False   
4     False           False                        False  False   
..      ...             ...                          ...    ...   
995   False           False                        False  False   
996   False           False                        False  False   
997   False           False                        False  False   
998   False           False                        False  False   
999   False           False                        False  False   

     test preparation course  math score  reading score  writing score  
0                      False       False          False          False  
1                      False       False          False          False  
2                      False       False          False          False  
3                      False       False          False          False  
4                      False       False          False          False  
..                       ...         ...            ...            ...  
995                    False       False          False          False  
996                    False       False          False          False  
997                    False       False          False          False  
998                    False       False          False          False  
999                    False       False          False          False  

[1000 rows x 8 columns]

# 1.Total number of males/females participated in the test

In [33]:
df['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

No: of males participated in the test=482
 No: of females participated in the test =518 

# 2.Count of various levels of parental education

In [44]:
df['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

# 3.(1)Based on gender who scores most on average in maths,reading and writing

In [20]:
df1=df.groupby('gender')['math score','reading score','writing score'].mean()

In [14]:
df1

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

Females score more in tests

# 3(2).Based on the test preparation who scores  most on average?

In [21]:
df2=df.groupby('test preparation course')['math score','reading score','writing score'].mean()

In [18]:
df2

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

Students who completed test preparation course score more

# 4.Variation in tests based on gender and test preparation course

In [46]:
data_std=df.groupby(['gender'])[['math score','reading score','writing score']].std()

In [47]:
data_std

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

In [49]:
data1_std=df.groupby(['test preparation course'])[['math score','reading score','writing score']].std()

In [50]:
data1_std

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

# 5.Students scored highest in maths

In [42]:
data=df.nlargest(250,['math score'])

In [43]:
data

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
856    male        group E            some high school      standard   
855    male        group E          associate's degree  free/reduced   
854    male        group D            some high school      standard   
853    male        group D          associate's degree  free/reduced   
852    male        group D                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
856               completed          77             76             77  
855               completed          77             69             68  
854               completed          77             68             69  
853                    none          77             78             73  
852                    none          77             62             64  

[250 rows x 8 columns]